app/scripts/services/Room.js

```diff
@@ -1,12 +1,21 @@
+// remember Room.js is a service. we don't use 'this' to refer to 'Room' like we would in a controller. 
 (function() {
     function Room($firebaseArray) {
-        
+        var Room = {};
         var ref = firebase.database().ref().child("rooms");
+        //get array of rooms stored in database on firebase
         var rooms = $firebaseArray(ref);
         
-        return {
-            all: rooms
+        // makes 'rooms' array public. used in home.html
+        Room.all = rooms;
+        
+        // to add newly created 'room' to the 'rooms' array.
+        //call in modalCtrl.js
+        Room.add = function (room) {
+            rooms.$add(room);

```

Comment:
I just want to make sure you understand you are passing an object here, which is correct. This evaluates as:

```
rooms.$add({ name: 'some_new_name' });
```